lib/dal/src/context.rs

```diff
@@ -979,16 +979,12 @@ impl DalContext {
         new
     }
 
-    async fn workspace(&self) -> TransactionsResult<Workspace> {
```

Comment:
One semantic difference here: when you have no workspace in your DalContext, get_workspace() will query for WorkspacePk::NONE and hopefully yield `WorkspaceError::WorkspaceNotFound`, while workspace() will return `TenancyError::NoWorkspace`. Is this change intentionally doing this differently or is it just that we know it'll probably eventually error in either case (assuming NONE is not in the db)?

Reply:
ok i've kept the two functions and made their behavior more clear.